playground/lgsvl/tc6/main.py

```diff
@@ -1,106 +1,155 @@
-    from lgsvl.agent import Agent
-
-    sim = connect_simulation("127.0.0.1", 8181)
-    load_scene(sim, TEST_PLACE.map_name)
-
-    pedestrian_behavior = _generate_initial_pedestrian_behavior()
-    pedestrian = load_pedestrian(sim, TEST_PLACE.map_name, pedestrian_behavior.initial_state)
-    pedestrian.follow(pedestrian_behavior.waypoints)
-
-    initial_ego_state, time_to_crash_point = _generate_initial_ego_state(pedestrian_behavior)
-    ego = load_ego(sim, "Jaguar2015XE (Apollo 5.0, many sensors)", initial_ego_state)
-
-    test_result = _TestResult()
-
-    def _on_collision(agent1: Agent, agent2: Agent, contact: Vector) -> None:
-        test_result.successful = False
-        sim.stop()
-
-    detect_collisions(sim, _on_collision)
-
-    sim.run(time_to_crash_point * 2)
-
-    # FIXME Sometimes the following print is not visible on the console (but being in debug mode)
-    print("Test succeeded" if test_result.successful else "Test failed")
-
-
-if __name__ == "__main__":
-    _main()
+import unittest
+from sys import stdout
+from typing import Callable
+
+from lgsvl import Simulator
+
+from tc6.config import TestConfig
+from tc6.testcase6 import TestCase6
+from tc6.locations import ALL_LOCATIONS, Location
+
+LGSVL_HOST: str = "127.0.0.1"
+LGSVL_PORT: int = 8181
+LOGGING_CONFIG = {
+    "version": 1,
+    "handlers": {
+        "console": {
+            "class": "logging.StreamHandler",
+            "stream": stdout
+        }
+    },
+    "root": {
+        "handlers": ["console"],
+        "level": "DEBUG"
+    }
+}
+
+
+class TC6TestSuite(unittest.TestCase):
+    _sim: Simulator = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        from common.simulator import connect_simulation
+        import logging.config
+        TC6TestSuite._sim = connect_simulation(LGSVL_HOST, LGSVL_PORT)
+        logging.config.dictConfig(LOGGING_CONFIG)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        TC6TestSuite._sim.stop()
+
+    def _iterate_configs(self, config_provider: Callable[[Location, bool], TestConfig]) -> None:
+        import logging
+        from inspect import currentframe, getouterframes
+
+        current_frame = currentframe()
+        calling_frame = getouterframes(current_frame, 2)
+        logging.info("Start {}".format(calling_frame[1][3]))
+
+        all_succeeded = True
+        for location in ALL_LOCATIONS:
+            for pedestrian_direction in [True, False]:
+                config = config_provider(location, pedestrian_direction)
+                test_result = TestCase6.execute(TC6TestSuite._sim, config)
+                if test_result is None:
+                    logging.warning("Skipped config {}".format(config))
+                else:
+                    if not test_result:
+                        logging.debug("Failed with config {}".format(config))
+                        all_succeeded = False
+        self.assertTrue(all_succeeded)
```

Comment:
It could be better if we have a message parameter next to the var all_succeeded. By looking at the test name, it is hard to guess what the test is doing and its expectation, so a message in this place will be useful.